internal/fullnode/build_pods.go

```diff
@@ -22,6 +22,8 @@ func BuildPods(crd *cosmosv1.CosmosFullNode, cksums ConfigChecksums) ([]diff.Res
 		if err != nil {
 			return nil, err
 		}
+		pod.Spec.Hostname = pod.Name
+		pod.Spec.Subdomain = crd.Name
```

Comment:
Please move to pod_builder.go. Otherwise, add unit test coverage showing why these changes are needed.